kura/org.eclipse.kura.linux.watchdog/src/main/java/org/eclipse/kura/linux/watchdog/WatchdogServiceImpl.java

```diff
@@ -338,6 +319,29 @@ private void readProperties(Map<String, Object> properties) {
         }
     }
 
+    private void cancelPollThread() {
+        if (pollThreadTask != null && !pollThreadTask.isCancelled()) {
+            logger.debug("Cancelling WatchdogServiceImpl task ...");
+            pollThreadTask.cancel(true);
```

Comment:
If you shut down the executor and the task was scheduled with this executor, then there is no need to cancel the task itself.